ibis/backends/flink/tests/test_ddl.py

```diff
@@ -262,19 +262,38 @@ def test_force_recreate_in_mem_table(
         assert new_table.schema() == schema
 
 
-def test_create_source_table_with_watermark(
-    con, functional_alltypes_schema, temp_table, csv_source_configs
+@pytest.mark.parametrize("primary_keys", [None, {"id"}, {"id", "string_col"}])
+def test_create_source_table_with_watermark_and_primary_keys(
+    con, functional_alltypes_schema, temp_table, csv_source_configs, primary_keys
```

Comment:
Is `functional_alltypes_schema` not needed here? What changed to specify it explicitly instead?

Reply:
> Is functional_alltypes_schema not needed here?

Yes, `functional_alltypes_schema` is not needed, removed it from the args.

> What changed to specify it explicitly instead?

Primary keys must be `non-nullable` in Flink:
> Flink will assume correctness of the primary key by assuming that the columns nullability is aligned with the columns in primary key. Connectors should ensure those are aligned.

For the following reason:
> Notes: In a CREATE TABLE statement, creating a primary key constraint will alter the columns nullability, that means, a column with primary key constraint is not nullable.
`new_table.schema()` returns nullable field types for primary-key columns, which leads to an assertion on this line:
```
assert new_table.schema() == schema
```
This is why I ended up defining a copy of `functional_alltypes_schema` with `"id": dt.int32(nullable=False)` and `"string_col": dt.string(nullable=False)`.